### PR TITLE
Add protocol stubs, make JSON specification const

### PR DESCRIPTION
**Pull request by Schwusch**
I've added stubs of all the service protocols declared in `aws-sdk-js`. I've also added the protocol calls in `library_builder.dart`.

### CODE REVIEW

File: aws_client/lib/src/json/builder.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import 'dart:convert';
+
+class JsonBuilder {
+  String build(Map<String, dynamic> value, Map<String, dynamic> shape) =>
+      jsonEncode(_translate(value, shape));
+
+  dynamic _translate(Map<String, dynamic> value, Map<String, dynamic> shape) {
```
Comment (isoos) [inline aws_client/lib/src/json/builder.dart]: I'm not sure I understand the need for this. Could you please link into the JS implementation what this supposed to be doing?

File: generator/lib/library_builder.dart [hunk 2]
```diff
@@ -206,10 +205,8 @@ class $className {""");
 
     writeln("""
     buildRequest(
-      RequestWrapper(
         spec,
-        ${parameterMembers.isNotEmpty ? "param.toJson()..addAll({'operation': '$methodName'})," : "{'operation': '$methodName'}"}
-      ),
+        ${parameterMembers.isNotEmpty ? "param.toJson()..addAll({'operation': '$methodName'})" : "{'operation': '$methodName'}"},
```
Comment (isoos) [inline generator/lib/library_builder.dart]: Maybe simpler:
```dart
{
  ...param.toJson(),
  'operation': methodName,
}
```

**Comment by isoos** [comment]
Merging this for now, let's iterate on it as we go...

File: generator/lib/library_builder.dart (hunk 2 above)
Comment (Schwusch) [reply generator/lib/library_builder.dart]: I did not know that was possible, nice catch

File: aws_client/lib/src/json/builder.dart (hunk 1 above)
Comment (Schwusch) [reply aws_client/lib/src/json/builder.dart]: Sure, https://github.com/aws/aws-sdk-js/blob/eaf5f577b7b719c055aa72f6e40cf606fb72c34e/lib/json/builder.js#L3

**Pull request merged**